DesktopEditor/common/File.cpp

```diff
@@ -1522,6 +1526,22 @@ namespace NSFile
         return sRet;
 #endif
 
+#if defined(__FreeBSD__)
+        int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+        char buf[NS_FILE_MAX_PATH];
+        size_t size = NS_FILE_MAX_PATH;
+
+        memset(buf, 0, NS_FILE_MAX_PATH);
+        if (sysctl(mib, 4, &buf, &size, NULL, 0) != 0) {
+            size = readlink("/proc/curproc/file", buf, size - 1);
```

Comment:
/proc is not mounted by default, not sure it's worth keeping it.

Reply:
You can have /proc mounted. sysctl might be enough for most of the cases but if not (I don't know in which case it wouldn't work..) you could still mount /proc and keep that code working

Reply:
As you want, I never saw a sysctl failing and if it's the case you're probably doomed somehow.